distributed/preloading.py

```diff
@@ -222,13 +222,34 @@ async def teardown(self):
                 await future
 
 
+class PreloadManager:
+    _preloads: list[Preload]
+
+    def __init__(self, preloads: list[Preload]):
+        self._preloads = preloads
+
+    async def start(self):
+        for preload in self._preloads:
+            try:
+                await preload.start()
+            except Exception:
+                logger.exception("Failed to start preload: %s", preload.name)
+
+    async def teardown(self):
+        for preload in self._preloads:
```

Comment:
We may want to handle preloads as a stack or gather on start/teardown.

Reply:
-0.5 from me. I would not want to break user-defined preloads in case they depend on sequentiality, and the benefit feels very mild to me.